--- a/arch/arm64/boot/dts/freescale/moment-tqmls1088a-connect.dts
+++ b/arch/arm64/boot/dts/freescale/moment-tqmls1088a-connect.dts
@@ -13,6 +13,7 @@
 		serial1 = "/soc/serial@21c0600";
 		serial2 = "/soc/serial@21d0500";
 		serial3 = "/soc/serial@21d0600";
+		gpio_expander = &pca9555;
 	};
 
 	ram {
@@ -419,6 +420,14 @@
 		};
 	};
 
+	vcc_3v3: regulator-vcc-3v3 {
+		compatible = "regulator-fixed";
+		regulator-name = "3V3";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+	};
+
+
 	timer {
 		compatible = "arm,armv8-timer";
 		interrupts = <0x01 0x0d 0x08 0x01 0x0e 0x08 0x01 0x0b 0x08 0x01 0x0a 0x08>;
@@ -664,6 +673,39 @@
 				reg = <0x57>;
 			};
 
+			at24cs01@54 {
+				compatible = "atmel,24c01";
+				label = "mb-eeprom";
+				reg = <0x54>;
+			};
+			at24cs01@5c {
+				compatible = "atmel,24cs01";
+				label = "mb-uuid";
+				reg = <0x5c>;
+			};
+
+			at24cs01@55 {
+				compatible = "atmel,24c01";
+				label = "db-eeprom";
+				reg = <0x55>;
+			};
+			at24cs01@5d {
+				compatible = "atmel,24cs01";
+				label = "db-uuid";
+				reg = <0x5d>;
+			};
+
+			at24cs01@52 {
+				compatible = "atmel,24c01";
+				label = "ant-eeprom";
+				reg = <0x52>;
+			};
+			at24cs01@5a {
+				compatible = "atmel,24cs01";
+				label = "ant-uuid";
+				reg = <0x5a>;
+			};
+
 			pca9555: pca9555@20 {
 				compatible = "nxp,pca9555";
 				gpio-controller;
@@ -671,6 +713,31 @@
 				reg = <0x20>;
 				gpio-base = <0>;
 			};
+
+			ads7830@4a {
+				label = "mb-sensor-adc";
+				compatible = "ti,ads7830";
+				reg = <0x4a>;
+			};
+
+			ads7830@4b {
+				label = "db-sensor-adc";
+				compatible = "ti,ads7830";
+				reg = <0x4b>;
+				vref-supply = <&vcc_3v3>;
+			};
+
+			tmp1075@48 {
+				label = "mb-sensor-temperature";
+				compatible = "ti,tmp1075";
+				reg = <0x48>;
+			};
+
+			tmp1075@49 {
+				label = "db-sensor-temperature";
+				compatible = "ti,tmp1075";
+				reg = <0x49>;
+			};
 		};
 
 		i2c@2010000 {
@@ -701,59 +768,6 @@
 			interrupts = <0x00 0x23 0x04>;
 			clocks = <0x02 0x04 0x07>;
 			status = "okay";
-
-			i2c-mux@70 {
-				compatible = "nxp,pca9544";
-				reg = <0x70>;
-				#address-cells = <0x01>;
-				#size-cells = <0x00>;
-
-				i2c@0 {
-					#address-cells = <0x01>;
-					#size-cells = <0x00>;
-					reg = <0x00>;
-
-					pca9555@20 {
-						compatible = "nxp,pca9555";
-						reg = <0x20>;
-						gpio-controller;
-						#gpio-cells = <0x02>;
-					};
-
-					pca9555@21 {
-						compatible = "nxp,pca9555";
-						reg = <0x21>;
-						gpio-controller;
-						#gpio-cells = <0x02>;
-					};
-
-					pca9555@22 {
-						compatible = "nxp,pca9555";
-						reg = <0x22>;
-						gpio-controller;
-						#gpio-cells = <0x02>;
-						phandle = <0x14>;
-					};
-				};
-
-				i2c@1 {
-					#address-cells = <0x01>;
-					#size-cells = <0x00>;
-					reg = <0x01>;
-				};
-
-				i2c@2 {
-					#address-cells = <0x01>;
-					#size-cells = <0x00>;
-					reg = <0x02>;
-				};
-
-				i2c@3 {
-					#address-cells = <0x01>;
-					#size-cells = <0x00>;
-					reg = <0x03>;
-				};
-			};
 		};
 
 		esdhc@2140000 {
@@ -837,49 +851,39 @@
 					#address-cells = <0x01>;
 					#size-cells = <0x01>;
 
-					rcwpbi@0 {
-						label = "RCW/PBI";
-						reg = <0x00 0x10000>;
+					rcwpbi@00000000 {
+						label = "rcwpbi";
+						reg = <0x00000000 0x00100000>;
 					};
 
-					uboot@10000 {
-						label = "U-Boot";
-						reg = <0x10000 0xc0000>;
+					uboot@00100000 {
+						label = "uboot";
+						reg = <0x00100000 0x00300000>;
 					};
 
-					env@d0000 {
-						label = "U-Boot Environment";
-						reg = <0xd0000 0x10000>;
+					env@00400000 {
+						label = "env";
+						reg = <0x00400000 0x00020000>;
 					};
 
-					dpaadpl@e0000 {
-						label = "DPAA2 DPL";
-						reg = <0xe0000 0x10000>;
+					dpaadpl@00420000 {
+						label = "envbackup";
+						reg = <0x00420000 0x00020000>;
 					};
 
-					dpaadpc@f0000 {
-						label = "DPAA2 DPC";
-						reg = <0xf0000 0x10000>;
+					dpaadpc@00500000 {
+						label = "mcfw";
+						reg = <0x00500000 0x00300000>;
 					};
 
-					mcfw@100000 {
-						label = "DPAA2 MC Firmware";
-						reg = <0x100000 0x1f0000>;
+					mcfw@00800000 {
+						label = "dpaadpc";
+						reg = <0x00800000 0x00100000>;
 					};
 
-					dtb@2f0000 {
-						label = "DTB";
-						reg = <0x2f0000 0x10000>;
-					};
-
-					linux@300000 {
-						label = "Linux";
-						reg = <0x300000 0x1500000>;
-					};
-
-					rootfs@1800000 {
-						label = "RootFS";
-						reg = <0x1800000 0x2800000>;
+					dtb@00900000 {
+						label = "dpaadpl";
+						reg = <0x00900000 0x00100000>;
 					};
 				};
 			};
@@ -1217,32 +1221,31 @@
 		};
 	};
 
-	gpio-keys-polled {
-		compatible = "gpio-keys-polled";
-		gpio-keys,name = "gpio-keys";
-		poll-interval = <0x64>;
-		autorepeat;
-
-		button0 {
-			label = "button0";
-			gpios = <0x14 0x05 0x01>;
-			linux,code = <0x3b>;
-		};
-
-		button1 {
-			label = "button1";
-			gpios = <0x14 0x06 0x01>;
-			linux,code = <0x3c>;
-		};
-	};
-
 	leds {
 		compatible = "gpio-leds";
 
-		user {
-			gpios = <0x14 0x0d 0x01>;
-			label = "led:user";
-			linux,default-trigger = "heartbeat";
+		mcu_boot0 {
+			label = "mcu-boot0";
+			gpios = <&pca9555 8 0x00>;
+			default-state = "off";
+		};
+
+		mcu_nrst {
+			label = "mcu-nrst";
+			gpios = <&pca9555 9 0x00>;
+			default-state = "on";
+		};
+
+		adcen_bat {
+			label = "adcen-bat";
+			gpios = <&pca9555 10 0x00>;
+			default-state = "off";
+		};
+
+		adcen_edlc {
+			label = "adcen-edlc";
+			gpios = <&pca9555 11 0x00>;
+			default-state = "off";
 		};
 	};
 
