--- a/drivers/net/phy/mscc/mscc.h
+++ b/drivers/net/phy/mscc/mscc.h
@@ -154,6 +154,8 @@ enum rgmii_clock_delay {
 #define FORCE_MDI_CROSSOVER_MDIX	  0x000C
 #define FORCE_MDI_CROSSOVER_MDI		  0x0008
 
+#define SIGDET_ACTIVE_LOW                 0x0001
+
 #define MSCC_PHY_ACTIPHY_CNTL		  20
 #define PHY_ADDR_REVERSED		  0x0200
 #define DOWNSHIFT_CNTL_MASK		  0x001C
--- a/drivers/net/phy/mscc/mscc_main.c
+++ b/drivers/net/phy/mscc/mscc_main.c
@@ -225,6 +225,9 @@ static int vsc85xx_mdix_set(struct phy_d
 	else if (mdix == ETH_TP_MDI_X)
 		reg_val = FORCE_MDI_CROSSOVER_MDIX;
 
+	// Change sigdet polarity
+	reg_val |= SIGDET_ACTIVE_LOW;
+
 	rc = phy_modify_paged(phydev, MSCC_PHY_PAGE_EXTENDED,
 			      MSCC_PHY_EXT_MODE_CNTL, FORCE_MDI_CROSSOVER_MASK,
 			      reg_val);
