--- a/arch/arm/cpu/armv8/fsl-layerscape/fsl_lsch3_serdes.c
+++ b/arch/arm/cpu/armv8/fsl-layerscape/fsl_lsch3_serdes.c
@@ -655,7 +655,7 @@ static void restart_clocks_and_reloc(voi
 void fsl_serdes_init(void)
 {
 #ifdef CONFIG_TQMLS1088A_BB_MOMENT_CONNECT
-    restart_clocks_and_reloc();
+	restart_clocks_and_reloc();
 #endif
 
 #if defined(CONFIG_FSL_MC_ENET) && !defined(CONFIG_SPL_BUILD)
@@ -712,6 +712,11 @@ void fsl_serdes_init(void)
 		    FSL_CHASSIS3_SRDS3_PRTCL_SHIFT,
 		    serdes3_prtcl_map);
 #endif
+
+#ifdef CONFIG_TQMLS1088A_BB_MOMENT_CONNECT
+	// Disable unused DPMAC
+	wriop_disable_dpmac(1);
+#endif
 }
 
 int serdes_set_env(int sd, int rcwsr, int sd_prctl_mask, int sd_prctl_shift)
--- a/board/tqc/tqmls1088a/tqmls1088a_bb_moment_connect.c
+++ b/board/tqc/tqmls1088a/tqmls1088a_bb_moment_connect.c
@@ -274,7 +274,7 @@ static int fdt_set_phy_handle(void *fdt,
 		return offset;
 
 	/* Get relevant dpmac node */
-	sprintf(name, "dpmac@%x", mac);
+	sprintf(name, "ethernet@%x", mac);
 	fdt_for_each_subnode(offset, fdt, offset) {
 		path = fdt_get_name(fdt, offset, &lenp);
 		if(!strncmp(name, path, strlen(name))) {
@@ -304,7 +304,7 @@ static int fdt_set_phy_mode(void *fdt, i
 		return offset;
 
 	/* Get relevant dpmac node */
-	sprintf(name, "dpmac@%x", mac);
+	sprintf(name, "ethernet@%x", mac);
 	fdt_for_each_subnode(offset, fdt, offset) {
 		path = fdt_get_name(fdt, offset, &lenp);
 		if(!strncmp(name, path, strlen(name))) {
