--- a/arch/arm/cpu/armv8/fsl-layerscape/fsl_lsch3_serdes.c
+++ b/arch/arm/cpu/armv8/fsl-layerscape/fsl_lsch3_serdes.c
@@ -639,16 +639,32 @@ static void restart_clocks_and_reloc(voi
 #endif
 
 	/* Check PLL lock status */
-	for(int serdes=0; serdes < 2; serdes++) {
-		struct ccsr_serdes __iomem *serdes_base = (serdes==0) ? serdes1_base : serdes2_base;
-		for(int pll=0; pll<2; pll++) {
-			uint32_t status = serdes_base->bank[pll].rstctl;
-			if (status&(1<<29))
-				printf("WARNING : Serdes %d PLL %d did not lock\n", serdes+1, pll+1);
-			else
-				printf("Serdes %d PLL %d locked\n", serdes+1, pll+1);
-		}
-	}
+        u8 status = get_pll_status();
+        for (int pll=0; pll<4; pll++) {
+                if (status & (1<<pll))
+                        printf("Serdes %d PLL %d locked\n", (pll>>1)+1, (pll&1)+1);
+                else
+                        printf("WARNING : Serdes %d PLL %d did not lock\n", (pll>>1)+1, (pll&1)+1);
+        }
+}
+
+u8 get_pll_status(void) {
+        struct ccsr_serdes __iomem *serdes1_base = (void *)CONFIG_SYS_FSL_LSCH3_SERDES_ADDR;
+        struct ccsr_serdes __iomem *serdes2_base = (void *)(CONFIG_SYS_FSL_LSCH3_SERDES_ADDR + 0x10000);
+        u8 bit = 0;
+        u8 pll_status = 0;
+
+        for (int serdes=0; serdes<2; serdes++) {
+                struct ccsr_serdes __iomem *serdes_base = (serdes==0) ? serdes1_base : serdes2_base;
+                for (int pll=0; pll<2; pll++) {
+                        uint32_t status = serdes_base->bank[pll].rstctl;
+                        if ((status&(1<<29)) == 0)
+                                pll_status |= (1<<bit);
+                        bit++;
+                }
+        }
+
+	return pll_status;
 }
 #endif
 
--- a/arch/arm/include/asm/arch-fsl-layerscape/fsl_serdes.h
+++ b/arch/arm/include/asm/arch-fsl-layerscape/fsl_serdes.h
@@ -215,4 +215,8 @@ int set_serdes_volt(int svdd);
 int setup_serdes_volt(u32 svdd);
 #endif
 
+#ifdef CONFIG_TQMLS1088A_BB_MOMENT_CONNECT
+u8 get_pll_status(void);
+#endif
+
 #endif /* __FSL_SERDES_H__ */
--- /dev/null
+++ b/board/tqc/common/Si5341-RevB-Registers.h
@@ -0,0 +1,728 @@
+/*
+ * Si5341 Rev B Configuration Register Export Header File
+ *
+ * This file represents a series of Silicon Labs Si5341 Rev B 
+ * register writes that can be performed to load a single configuration 
+ * on a device. It was created by a Silicon Labs ClockBuilder Pro
+ * export tool.
+ *
+ * Part:		                                       Si5341 Rev B
+ * Design ID:                                          
+ * Includes Pre/Post Download Control Register Writes: Yes
+ * Created By:                                         ClockBuilder Pro v2.10 [2016-09-07]
+ * Timestamp:                                          2023-10-20 17:38:31 GMT+02:00
+ *
+ * A complete design report corresponding to this export is included at the end 
+ * of this header file.
+ *
+ */
+
+#ifndef SI5341_REVB_REG_CONFIG_HEADER
+#define SI5341_REVB_REG_CONFIG_HEADER
+
+#define SI5341_REVB_REG_CONFIG_NUM_REGS				281
+
+typedef struct
+{
+	unsigned int address; /* 16-bit register address */
+	unsigned char value; /* 8-bit register data */
+
+} si5341_revb_register_t;
+
+si5341_revb_register_t const si5341_revb_registers[SI5341_REVB_REG_CONFIG_NUM_REGS] =
+{
+	{ 0x0B24, 0xD8 },
+	{ 0x0B25, 0x00 },
+	{ 0x000B, 0x74 },
+	{ 0x0017, 0x3A },
+	{ 0x0018, 0xFF },
+	{ 0x0021, 0x0F },
+	{ 0x0022, 0x00 },
+	{ 0x002B, 0x02 },
+	{ 0x002C, 0x20 },
+	{ 0x002D, 0x00 },
+	{ 0x002E, 0x00 },
+	{ 0x002F, 0x00 },
+	{ 0x0030, 0x00 },
+	{ 0x0031, 0x00 },
+	{ 0x0032, 0x00 },
+	{ 0x0033, 0x00 },
+	{ 0x0034, 0x00 },
+	{ 0x0035, 0x00 },
+	{ 0x0036, 0x00 },
+	{ 0x0037, 0x00 },
+	{ 0x0038, 0x00 },
+	{ 0x0039, 0x00 },
+	{ 0x003A, 0x00 },
+	{ 0x003B, 0x00 },
+	{ 0x003C, 0x00 },
+	{ 0x003D, 0x00 },
+	{ 0x0041, 0x00 },
+	{ 0x0042, 0x00 },
+	{ 0x0043, 0x00 },
+	{ 0x0044, 0x00 },
+	{ 0x009E, 0x00 },
+	{ 0x0102, 0x01 },
+	{ 0x0108, 0x02 },
+	{ 0x0109, 0x09 },
+	{ 0x010A, 0x3B },
+	{ 0x010B, 0x00 },
+	{ 0x010D, 0x06 },
+	{ 0x010E, 0x09 },
+	{ 0x010F, 0x3B },
+	{ 0x0110, 0x01 },
+	{ 0x0112, 0x02 },
+	{ 0x0113, 0x09 },
+	{ 0x0114, 0x3B },
+	{ 0x0115, 0x00 },
+	{ 0x0117, 0x02 },
+	{ 0x0118, 0x09 },
+	{ 0x0119, 0x3B },
+	{ 0x011A, 0x00 },
+	{ 0x011C, 0x02 },
+	{ 0x011D, 0x09 },
+	{ 0x011E, 0x3B },
+	{ 0x011F, 0x00 },
+	{ 0x0121, 0x02 },
+	{ 0x0122, 0x09 },
+	{ 0x0123, 0x3B },
+	{ 0x0124, 0x00 },
+	{ 0x0126, 0x02 },
+	{ 0x0127, 0x09 },
+	{ 0x0128, 0x3B },
+	{ 0x0129, 0x00 },
+	{ 0x012B, 0x02 },
+	{ 0x012C, 0x09 },
+	{ 0x012D, 0x3B },
+	{ 0x012E, 0x00 },
+	{ 0x0130, 0x01 },
+	{ 0x0131, 0x09 },
+	{ 0x0132, 0x3B },
+	{ 0x0133, 0x00 },
+	{ 0x013A, 0x01 },
+	{ 0x013B, 0x09 },
+	{ 0x013C, 0x3B },
+	{ 0x013D, 0x00 },
+	{ 0x013F, 0x00 },
+	{ 0x0140, 0x00 },
+	{ 0x0141, 0x40 },
+	{ 0x0202, 0x00 },
+	{ 0x0203, 0x00 },
+	{ 0x0204, 0x00 },
+	{ 0x0205, 0x00 },
+	{ 0x0206, 0x00 },
+	{ 0x0208, 0x00 },
+	{ 0x0209, 0x00 },
+	{ 0x020A, 0x00 },
+	{ 0x020B, 0x00 },
+	{ 0x020C, 0x00 },
+	{ 0x020D, 0x00 },
+	{ 0x020E, 0x00 },
+	{ 0x020F, 0x00 },
+	{ 0x0210, 0x00 },
+	{ 0x0211, 0x00 },
+	{ 0x0212, 0x00 },
+	{ 0x0213, 0x00 },
+	{ 0x0214, 0x00 },
+	{ 0x0215, 0x00 },
+	{ 0x0216, 0x00 },
+	{ 0x0217, 0x00 },
+	{ 0x0218, 0x00 },
+	{ 0x0219, 0x00 },
+	{ 0x021A, 0x00 },
+	{ 0x021B, 0x00 },
+	{ 0x021C, 0x00 },
+	{ 0x021D, 0x00 },
+	{ 0x021E, 0x00 },
+	{ 0x021F, 0x00 },
+	{ 0x0220, 0x00 },
+	{ 0x0221, 0x00 },
+	{ 0x0222, 0x00 },
+	{ 0x0223, 0x00 },
+	{ 0x0224, 0x00 },
+	{ 0x0225, 0x00 },
+	{ 0x0226, 0x00 },
+	{ 0x0227, 0x00 },
+	{ 0x0228, 0x00 },
+	{ 0x0229, 0x00 },
+	{ 0x022A, 0x00 },
+	{ 0x022B, 0x00 },
+	{ 0x022C, 0x00 },
+	{ 0x022D, 0x00 },
+	{ 0x022E, 0x00 },
+	{ 0x022F, 0x00 },
+	{ 0x0235, 0x00 },
+	{ 0x0236, 0x00 },
+	{ 0x0237, 0x00 },
+	{ 0x0238, 0xC0 },
+	{ 0x0239, 0xDA },
+	{ 0x023A, 0x00 },
+	{ 0x023B, 0x00 },
+	{ 0x023C, 0x00 },
+	{ 0x023D, 0x00 },
+	{ 0x023E, 0xC0 },
+	{ 0x024A, 0x03 },
+	{ 0x024B, 0x00 },
+	{ 0x024C, 0x00 },
+	{ 0x024D, 0x00 },
+	{ 0x024E, 0x00 },
+	{ 0x024F, 0x00 },
+	{ 0x0250, 0x04 },
+	{ 0x0251, 0x00 },
+	{ 0x0252, 0x00 },
+	{ 0x0253, 0x04 },
+	{ 0x0254, 0x00 },
+	{ 0x0255, 0x00 },
+	{ 0x0256, 0x04 },
+	{ 0x0257, 0x00 },
+	{ 0x0258, 0x00 },
+	{ 0x0259, 0x04 },
+	{ 0x025A, 0x00 },
+	{ 0x025B, 0x00 },
+	{ 0x025C, 0x03 },
+	{ 0x025D, 0x00 },
+	{ 0x025E, 0x00 },
+	{ 0x025F, 0x03 },
+	{ 0x0260, 0x00 },
+	{ 0x0261, 0x00 },
+	{ 0x0262, 0x00 },
+	{ 0x0263, 0x00 },
+	{ 0x0264, 0x00 },
+	{ 0x0268, 0x00 },
+	{ 0x0269, 0x00 },
+	{ 0x026A, 0x00 },
+	{ 0x026B, 0x00 },
+	{ 0x026C, 0x00 },
+	{ 0x026D, 0x00 },
+	{ 0x026E, 0x00 },
+	{ 0x026F, 0x00 },
+	{ 0x0270, 0x00 },
+	{ 0x0271, 0x00 },
+	{ 0x0272, 0x00 },
+	{ 0x0302, 0x00 },
+	{ 0x0303, 0x00 },
+	{ 0x0304, 0x00 },
+	{ 0x0305, 0x00 },
+	{ 0x0306, 0x07 },
+	{ 0x0307, 0x00 },
+	{ 0x0308, 0x00 },
+	{ 0x0309, 0x00 },
+	{ 0x030A, 0x00 },
+	{ 0x030B, 0x80 },
+	{ 0x030C, 0x00 },
+	{ 0x030D, 0x00 },
+	{ 0x030E, 0x00 },
+	{ 0x030F, 0x00 },
+	{ 0x0310, 0x00 },
+	{ 0x0311, 0x1C },
+	{ 0x0312, 0x00 },
+	{ 0x0313, 0x00 },
+	{ 0x0314, 0x00 },
+	{ 0x0315, 0x00 },
+	{ 0x0316, 0xA0 },
+	{ 0x0317, 0x00 },
+	{ 0x0318, 0x00 },
+	{ 0x0319, 0x00 },
+	{ 0x031A, 0x00 },
+	{ 0x031B, 0x00 },
+	{ 0x031C, 0x00 },
+	{ 0x031D, 0x00 },
+	{ 0x031E, 0x00 },
+	{ 0x031F, 0x00 },
+	{ 0x0320, 0x00 },
+	{ 0x0321, 0x00 },
+	{ 0x0322, 0x00 },
+	{ 0x0323, 0x00 },
+	{ 0x0324, 0x00 },
+	{ 0x0325, 0x00 },
+	{ 0x0326, 0x00 },
+	{ 0x0327, 0x00 },
+	{ 0x0328, 0x00 },
+	{ 0x0329, 0x00 },
+	{ 0x032A, 0x00 },
+	{ 0x032B, 0x00 },
+	{ 0x032C, 0x00 },
+	{ 0x032D, 0x00 },
+	{ 0x032E, 0x00 },
+	{ 0x032F, 0x00 },
+	{ 0x0330, 0x00 },
+	{ 0x0331, 0x00 },
+	{ 0x0332, 0x00 },
+	{ 0x0333, 0x00 },
+	{ 0x0334, 0x00 },
+	{ 0x0335, 0x00 },
+	{ 0x0336, 0x00 },
+	{ 0x0337, 0x00 },
+	{ 0x0338, 0x00 },
+	{ 0x0339, 0x1F },
+	{ 0x033B, 0x00 },
+	{ 0x033C, 0x00 },
+	{ 0x033D, 0x00 },
+	{ 0x033E, 0x00 },
+	{ 0x033F, 0x00 },
+	{ 0x0340, 0x00 },
+	{ 0x0341, 0x00 },
+	{ 0x0342, 0x00 },
+	{ 0x0343, 0x00 },
+	{ 0x0344, 0x00 },
+	{ 0x0345, 0x00 },
+	{ 0x0346, 0x00 },
+	{ 0x0347, 0x00 },
+	{ 0x0348, 0x00 },
+	{ 0x0349, 0x00 },
+	{ 0x034A, 0x00 },
+	{ 0x034B, 0x00 },
+	{ 0x034C, 0x00 },
+	{ 0x034D, 0x00 },
+	{ 0x034E, 0x00 },
+	{ 0x034F, 0x00 },
+	{ 0x0350, 0x00 },
+	{ 0x0351, 0x00 },
+	{ 0x0352, 0x00 },
+	{ 0x0353, 0x00 },
+	{ 0x0354, 0x00 },
+	{ 0x0355, 0x00 },
+	{ 0x0356, 0x00 },
+	{ 0x0357, 0x00 },
+	{ 0x0358, 0x00 },
+	{ 0x0359, 0x00 },
+	{ 0x035A, 0x00 },
+	{ 0x035B, 0x00 },
+	{ 0x035C, 0x00 },
+	{ 0x035D, 0x00 },
+	{ 0x035E, 0x00 },
+	{ 0x035F, 0x00 },
+	{ 0x0360, 0x00 },
+	{ 0x0361, 0x00 },
+	{ 0x0362, 0x00 },
+	{ 0x0802, 0x00 },
+	{ 0x0803, 0x00 },
+	{ 0x0804, 0x00 },
+	{ 0x090E, 0x02 },
+	{ 0x091C, 0x04 },
+	{ 0x0943, 0x01 },
+	{ 0x0949, 0x00 },
+	{ 0x094A, 0x00 },
+	{ 0x0A02, 0x00 },
+	{ 0x0A03, 0x03 },
+	{ 0x0A04, 0x01 },
+	{ 0x0A05, 0x03 },
+	{ 0x0B44, 0x0F },
+	{ 0x0B4A, 0x1C },
+	{ 0x001C, 0x01 },
+	{ 0x0B24, 0xDB },
+	{ 0x0B25, 0x02 },
+
+};
+
+/*
+ * Design Report
+ *
+ * Overview
+ * ========
+ * Part:         Si5341 Rev B
+ * Project File: Z:\ClockBuilder\clock\Si5341-RevB-Project.slabtimeproj
+ * Design ID:    <none>
+ * Created By:   ClockBuilder Pro v2.10 [2016-09-07]
+ * Timestamp:    2023-10-20 17:38:31 GMT+02:00
+ * 
+ * Design Rule Check
+ * =================
+ * Errors:
+ * - No errors
+ * 
+ * Warnings:
+ * - OUT1 [156.25 MHz] and OUT2 [100 MHz] may have harmonic crosstalk [1]
+ * - Revision B is not recommended for new designs
+ * 
+ * Notes:
+ * [1] To avoid crosstalk in outputs, Silicon Labs recommends the following:
+ * - Avoid adjacent frequency values that are close. CBPro uses an output's integration bandwidth (IBW) to determine whether two adjacent frequencies are too close. An IBW of 20 MHz is used for frequencies 80 MHz and larger. Lower frequencies will use IBW of Freq/4.
+ * - Note that harmonics of the fundamental clock output frequencies also contribute to crosstalk, but to a lesser degree than the fundamental.
+ * - Adjacent frequency values that are integer multiples of one another are okay and these outputs should be grouped accordingly. For example, a 155.52 MHz and 622.08 MHz (155.52 x 4) can be adjacent.
+ * - Unused outputs can be used to separate clock outputs that might otherwise interfere with one another.
+ * - Silicon Labs recommends you validate your design's jitter performance using an Evaluation Board.
+ * 
+ * Device Grade
+ * ============
+ * Maximum Output Frequency: 156.25 MHz
+ * Frequency Synthesis Mode: Fractional
+ * Frequency Plan Grade:     B
+ * Minimum Base OPN:         Si5341B*
+ * 
+ * Base       Output Clock         Supported Frequency Synthesis Modes
+ * OPN Grade  Frequency Range      (Typical Jitter)
+ * ---------  -------------------  --------------------------------------------
+ * Si5341A    100 Hz to 712.5 MHz  Integer (< 100 fs) and fractional (< 150 fs)
+ * Si5341B*   100 Hz to 350 MHz    "
+ * Si5341C    100 Hz to 712.5 MHz  Integer only (< 100 fs)
+ * Si5341D    100 Hz to 350 MHz    "
+ * 
+ * * Based on your calculated frequency plan, a Si5341B grade device is
+ * sufficient for your design. For more in-system configuration flexibility
+ * (higher frequencies and/or to enable fractional synthesis), consider
+ * selecting device grade Si5341A when specifying an ordering part number (OPN)
+ * for your application. See the datasheet Ordering Guide for more information.
+ * 
+ * Design
+ * ======
+ * Host Interface:
+ *    I/O Power Supply: VDDA (3.3V)
+ *    SPI Mode: 4-Wire
+ *    I2C Address Range: 116d to 119d / 0x74 to 0x77 (selected via A0/A1 pins)
+ * 
+ * Inputs:
+ *    XAXB: 48 MHz
+ *          Crystal Mode
+ *     IN0: Unused
+ *     IN1: Unused
+ *     IN2: Unused
+ *   FB_IN: Unused
+ * 
+ * Outputs:
+ *    OUT0: 125 MHz
+ *          Enabled, LVDS 2.5 V
+ *    OUT1: 156.25 MHz [ 156 + 1/4 MHz ]
+ *          Enabled, LVDS 2.5 V
+ *    OUT2: 100 MHz
+ *          Enabled, LVDS 2.5 V
+ *    OUT3: 100 MHz
+ *          Enabled, LVDS 2.5 V
+ *    OUT4: 100 MHz
+ *          Enabled, LVDS 2.5 V
+ *    OUT5: 100 MHz
+ *          Enabled, LVDS 2.5 V
+ *    OUT6: 125 MHz
+ *          Enabled, LVDS 2.5 V
+ *    OUT7: 125 MHz
+ *          Enabled, LVDS 2.5 V
+ *    OUT8: Unused
+ *    OUT9: Unused
+ * 
+ * Frequency Plan
+ * ==============
+ * Fpfd = 48 MHz
+ * Fvco = 14 GHz
+ * Fms0 = 1 GHz
+ * Fms1 = 312.5 MHz [ 312 + 1/2 MHz ]
+ * 
+ * P dividers:
+ *    P0  = Unused
+ *    P1  = Unused
+ *    P2  = Unused
+ *    P3  = Unused
+ *    Pxaxb = 1
+ * 
+ * M = 291.6666666666666666... [ 291 + 2/3 ]
+ * N dividers:
+ *    N0:
+ *       Value: 14
+ *       Skew:  0.000 s
+ *       OUT0: 125 MHz
+ *       OUT2: 100 MHz
+ *       OUT3: 100 MHz
+ *       OUT4: 100 MHz
+ *       OUT5: 100 MHz
+ *       OUT6: 125 MHz
+ *       OUT7: 125 MHz
+ *    N1:
+ *       Value: 44.8 [ 44 + 4/5 ]
+ *       Skew:  0.000 s
+ *       OUT1: 156.25 MHz [ 156 + 1/4 MHz ]
+ *    N2:
+ *       Unused
+ *    N3:
+ *       Unused
+ *    N4:
+ *       Unused
+ * 
+ * R dividers:
+ *    R0 = 8
+ *    R1 = 2
+ *    R2 = 10
+ *    R3 = 10
+ *    R4 = 10
+ *    R5 = 10
+ *    R6 = 8
+ *    R7 = 8
+ *    R8 = Unused
+ *    R9 = Unused
+ * 
+ * Dividers listed above show effective values. These values are translated to register settings by ClockBuilder Pro. For the actual register values, see below. Refer to the Family Reference Manual for information on registers related to frequency plan.
+ * 
+ * Digitally Controlled Oscillator (DCO)
+ * =====================================
+ * Mode: Register Direct Write
+ * 
+ * N0: DCO Disabled
+ * 
+ * N1: DCO Disabled
+ * 
+ * N2: DCO Disabled
+ * 
+ * N3: DCO Disabled
+ * 
+ * N4: DCO Disabled
+ * 
+ * Estimated Power & Junction Temperature
+ * ======================================
+ * Assumptions:
+ * 
+ * Revision: B
+ * VDD:      1.8 V
+ * Ta:       70 °C
+ * Airflow:  None
+ * 
+ * Total Power: 928 mW, On Chip Power: 880 mW, Tj: 89 °C
+ * 
+ *          Frequency  Format   Voltage   Current     Power
+ *         ----------  ------  --------  --------  --------
+ * VDD                            1.8 V  126.1 mA    227 mW
+ * VDDA                           3.3 V  117.4 mA    387 mW
+ * VDDO0      125 MHz  LVDS       2.5 V   15.7 mA     39 mW
+ * VDDO1   156.25 MHz  LVDS       2.5 V   15.9 mA     40 mW
+ * VDDO2      100 MHz  LVDS       2.5 V   15.6 mA     39 mW
+ * VDDO3      100 MHz  LVDS       2.5 V   15.6 mA     39 mW
+ * VDDO4      100 MHz  LVDS       2.5 V   15.6 mA     39 mW
+ * VDDO5      100 MHz  LVDS       2.5 V   15.6 mA     39 mW
+ * VDDO6      125 MHz  LVDS       2.5 V   15.7 mA     39 mW
+ * VDDO7      125 MHz  LVDS       2.5 V   15.7 mA     39 mW
+ * VDDO8       Unused                                      
+ * VDDO9       Unused                                      
+ *                                       --------  --------
+ *                                Total  368.8 mA    928 mW
+ * 
+ * Note:
+ * 
+ * -Total power includes on- and off-chip power. This is a typical value and estimate only.
+ * -Use an EVB for a more exact power measurement
+ * -On-chip power excludes power dissipated in external terminations.
+ * -Tj is junction temperature. When using a crystal (XTAL) reference, Tj must be less than 125 °C (on Si5341 Revision B) for device to comply with datasheet specifications.
+ * 
+ * Settings
+ * ========
+ * 
+ * Location      Setting Name         Decimal Value  Hex Value     
+ * ------------  -------------------  -------------  --------------
+ * 0x000B[6:0]   I2C_ADDR             116            0x74          
+ * 0x0017[0]     SYSINCAL_INTR_MSK    0              0x0           
+ * 0x0017[1]     LOSXAXB_INTR_MSK     1              0x1           
+ * 0x0017[2]     LOSREF_INTR_MSK      0              0x0           
+ * 0x0017[3]     LOL_INTR_MSK         1              0x1           
+ * 0x0017[5]     SMB_TMOUT_INTR_MSK   1              0x1           
+ * 0x0018[3:0]   LOSIN_INTR_MSK       15             0xF           
+ * 0x0021[0]     IN_SEL_REGCTRL       1              0x1           
+ * 0x0021[2:1]   IN_SEL               3              0x3           
+ * 0x0022[1]     OE                   0              0x0           
+ * 0x002B[3]     SPI_3WIRE            0              0x0           
+ * 0x002B[5]     AUTO_NDIV_UPDATE     0              0x0           
+ * 0x002C[3:0]   LOS_EN               0              0x0           
+ * 0x002C[4]     LOSXAXB_DIS          0              0x0           
+ * 0x002D[1:0]   LOS0_VAL_TIME        0              0x0           
+ * 0x002D[3:2]   LOS1_VAL_TIME        0              0x0           
+ * 0x002D[5:4]   LOS2_VAL_TIME        0              0x0           
+ * 0x002D[7:6]   LOS3_VAL_TIME        0              0x0           
+ * 0x002E[15:0]  LOS0_TRG_THR         0              0x0000        
+ * 0x0030[15:0]  LOS1_TRG_THR         0              0x0000        
+ * 0x0032[15:0]  LOS2_TRG_THR         0              0x0000        
+ * 0x0034[15:0]  LOS3_TRG_THR         0              0x0000        
+ * 0x0036[15:0]  LOS0_CLR_THR         0              0x0000        
+ * 0x0038[15:0]  LOS1_CLR_THR         0              0x0000        
+ * 0x003A[15:0]  LOS2_CLR_THR         0              0x0000        
+ * 0x003C[15:0]  LOS3_CLR_THR         0              0x0000        
+ * 0x0041[4:0]   LOS0_DIV_SEL         0              0x00          
+ * 0x0042[4:0]   LOS1_DIV_SEL         0              0x00          
+ * 0x0043[4:0]   LOS2_DIV_SEL         0              0x00          
+ * 0x0044[4:0]   LOS3_DIV_SEL         0              0x00          
+ * 0x009E[7:4]   LOL_SET_THR          0              0x0           
+ * 0x0102[0]     OUTALL_DISABLE_LOW   1              0x1           
+ * 0x0108[0]     OUT0_PDN             0              0x0           
+ * 0x0108[1]     OUT0_OE              1              0x1           
+ * 0x0108[2]     OUT0_RDIV_FORCE2     0              0x0           
+ * 0x0109[2:0]   OUT0_FORMAT          1              0x1           
+ * 0x0109[3]     OUT0_SYNC_EN         1              0x1           
+ * 0x0109[5:4]   OUT0_DIS_STATE       0              0x0           
+ * 0x0109[7:6]   OUT0_CMOS_DRV        0              0x0           
+ * 0x010A[3:0]   OUT0_CM              11             0xB           
+ * 0x010A[6:4]   OUT0_AMPL            3              0x3           
+ * 0x010B[2:0]   OUT0_MUX_SEL         0              0x0           
+ * 0x010B[7:6]   OUT0_INV             0              0x0           
+ * 0x010D[0]     OUT1_PDN             0              0x0           
+ * 0x010D[1]     OUT1_OE              1              0x1           
+ * 0x010D[2]     OUT1_RDIV_FORCE2     1              0x1           
+ * 0x010E[2:0]   OUT1_FORMAT          1              0x1           
+ * 0x010E[3]     OUT1_SYNC_EN         1              0x1           
+ * 0x010E[5:4]   OUT1_DIS_STATE       0              0x0           
+ * 0x010E[7:6]   OUT1_CMOS_DRV        0              0x0           
+ * 0x010F[3:0]   OUT1_CM              11             0xB           
+ * 0x010F[6:4]   OUT1_AMPL            3              0x3           
+ * 0x0110[2:0]   OUT1_MUX_SEL         1              0x1           
+ * 0x0110[7:6]   OUT1_INV             0              0x0           
+ * 0x0112[0]     OUT2_PDN             0              0x0           
+ * 0x0112[1]     OUT2_OE              1              0x1           
+ * 0x0112[2]     OUT2_RDIV_FORCE2     0              0x0           
+ * 0x0113[2:0]   OUT2_FORMAT          1              0x1           
+ * 0x0113[3]     OUT2_SYNC_EN         1              0x1           
+ * 0x0113[5:4]   OUT2_DIS_STATE       0              0x0           
+ * 0x0113[7:6]   OUT2_CMOS_DRV        0              0x0           
+ * 0x0114[3:0]   OUT2_CM              11             0xB           
+ * 0x0114[6:4]   OUT2_AMPL            3              0x3           
+ * 0x0115[2:0]   OUT2_MUX_SEL         0              0x0           
+ * 0x0115[7:6]   OUT2_INV             0              0x0           
+ * 0x0117[0]     OUT3_PDN             0              0x0           
+ * 0x0117[1]     OUT3_OE              1              0x1           
+ * 0x0117[2]     OUT3_RDIV_FORCE2     0              0x0           
+ * 0x0118[2:0]   OUT3_FORMAT          1              0x1           
+ * 0x0118[3]     OUT3_SYNC_EN         1              0x1           
+ * 0x0118[5:4]   OUT3_DIS_STATE       0              0x0           
+ * 0x0118[7:6]   OUT3_CMOS_DRV        0              0x0           
+ * 0x0119[3:0]   OUT3_CM              11             0xB           
+ * 0x0119[6:4]   OUT3_AMPL            3              0x3           
+ * 0x011A[2:0]   OUT3_MUX_SEL         0              0x0           
+ * 0x011A[7:6]   OUT3_INV             0              0x0           
+ * 0x011C[0]     OUT4_PDN             0              0x0           
+ * 0x011C[1]     OUT4_OE              1              0x1           
+ * 0x011C[2]     OUT4_RDIV_FORCE2     0              0x0           
+ * 0x011D[2:0]   OUT4_FORMAT          1              0x1           
+ * 0x011D[3]     OUT4_SYNC_EN         1              0x1           
+ * 0x011D[5:4]   OUT4_DIS_STATE       0              0x0           
+ * 0x011D[7:6]   OUT4_CMOS_DRV        0              0x0           
+ * 0x011E[3:0]   OUT4_CM              11             0xB           
+ * 0x011E[6:4]   OUT4_AMPL            3              0x3           
+ * 0x011F[2:0]   OUT4_MUX_SEL         0              0x0           
+ * 0x011F[7:6]   OUT4_INV             0              0x0           
+ * 0x0121[0]     OUT5_PDN             0              0x0           
+ * 0x0121[1]     OUT5_OE              1              0x1           
+ * 0x0121[2]     OUT5_RDIV_FORCE2     0              0x0           
+ * 0x0122[2:0]   OUT5_FORMAT          1              0x1           
+ * 0x0122[3]     OUT5_SYNC_EN         1              0x1           
+ * 0x0122[5:4]   OUT5_DIS_STATE       0              0x0           
+ * 0x0122[7:6]   OUT5_CMOS_DRV        0              0x0           
+ * 0x0123[3:0]   OUT5_CM              11             0xB           
+ * 0x0123[6:4]   OUT5_AMPL            3              0x3           
+ * 0x0124[2:0]   OUT5_MUX_SEL         0              0x0           
+ * 0x0124[7:6]   OUT5_INV             0              0x0           
+ * 0x0126[0]     OUT6_PDN             0              0x0           
+ * 0x0126[1]     OUT6_OE              1              0x1           
+ * 0x0126[2]     OUT6_RDIV_FORCE2     0              0x0           
+ * 0x0127[2:0]   OUT6_FORMAT          1              0x1           
+ * 0x0127[3]     OUT6_SYNC_EN         1              0x1           
+ * 0x0127[5:4]   OUT6_DIS_STATE       0              0x0           
+ * 0x0127[7:6]   OUT6_CMOS_DRV        0              0x0           
+ * 0x0128[3:0]   OUT6_CM              11             0xB           
+ * 0x0128[6:4]   OUT6_AMPL            3              0x3           
+ * 0x0129[2:0]   OUT6_MUX_SEL         0              0x0           
+ * 0x0129[7:6]   OUT6_INV             0              0x0           
+ * 0x012B[0]     OUT7_PDN             0              0x0           
+ * 0x012B[1]     OUT7_OE              1              0x1           
+ * 0x012B[2]     OUT7_RDIV_FORCE2     0              0x0           
+ * 0x012C[2:0]   OUT7_FORMAT          1              0x1           
+ * 0x012C[3]     OUT7_SYNC_EN         1              0x1           
+ * 0x012C[5:4]   OUT7_DIS_STATE       0              0x0           
+ * 0x012C[7:6]   OUT7_CMOS_DRV        0              0x0           
+ * 0x012D[3:0]   OUT7_CM              11             0xB           
+ * 0x012D[6:4]   OUT7_AMPL            3              0x3           
+ * 0x012E[2:0]   OUT7_MUX_SEL         0              0x0           
+ * 0x012E[7:6]   OUT7_INV             0              0x0           
+ * 0x0130[0]     OUT8_PDN             1              0x1           
+ * 0x0130[1]     OUT8_OE              0              0x0           
+ * 0x0130[2]     OUT8_RDIV_FORCE2     0              0x0           
+ * 0x0131[2:0]   OUT8_FORMAT          1              0x1           
+ * 0x0131[3]     OUT8_SYNC_EN         1              0x1           
+ * 0x0131[5:4]   OUT8_DIS_STATE       0              0x0           
+ * 0x0131[7:6]   OUT8_CMOS_DRV        0              0x0           
+ * 0x0132[3:0]   OUT8_CM              11             0xB           
+ * 0x0132[6:4]   OUT8_AMPL            3              0x3           
+ * 0x0133[2:0]   OUT8_MUX_SEL         0              0x0           
+ * 0x0133[7:6]   OUT8_INV             0              0x0           
+ * 0x013A[0]     OUT9_PDN             1              0x1           
+ * 0x013A[1]     OUT9_OE              0              0x0           
+ * 0x013A[2]     OUT9_RDIV_FORCE2     0              0x0           
+ * 0x013B[2:0]   OUT9_FORMAT          1              0x1           
+ * 0x013B[3]     OUT9_SYNC_EN         1              0x1           
+ * 0x013B[5:4]   OUT9_DIS_STATE       0              0x0           
+ * 0x013B[7:6]   OUT9_CMOS_DRV        0              0x0           
+ * 0x013C[3:0]   OUT9_CM              11             0xB           
+ * 0x013C[6:4]   OUT9_AMPL            3              0x3           
+ * 0x013D[2:0]   OUT9_MUX_SEL         0              0x0           
+ * 0x013D[7:6]   OUT9_INV             0              0x0           
+ * 0x013F[11:0]  OUTX_ALWAYS_ON       0              0x000         
+ * 0x0141[5]     OUT_DIS_LOL_MSK      0              0x0           
+ * 0x0141[7]     OUT_DIS_MSK_LOS_PFD  0              0x0           
+ * 0x0202[31:0]  XAXB_FREQ_OFFSET     0              0x00000000    
+ * 0x0206[1:0]   PXAXB                0              0x0           
+ * 0x0208[47:0]  P0                   0              0x000000000000
+ * 0x020E[31:0]  P0_SET               0              0x00000000    
+ * 0x0212[47:0]  P1                   0              0x000000000000
+ * 0x0218[31:0]  P1_SET               0              0x00000000    
+ * 0x021C[47:0]  P2                   0              0x000000000000
+ * 0x0222[31:0]  P2_SET               0              0x00000000    
+ * 0x0226[47:0]  P3                   0              0x000000000000
+ * 0x022C[31:0]  P3_SET               0              0x00000000    
+ * 0x0235[43:0]  M_NUM                939524096000   0x0DAC0000000 
+ * 0x023B[31:0]  M_DEN                3221225472     0xC0000000    
+ * 0x024A[23:0]  R0_REG               3              0x000003      
+ * 0x024D[23:0]  R1_REG               0              0x000000      
+ * 0x0250[23:0]  R2_REG               4              0x000004      
+ * 0x0253[23:0]  R3_REG               4              0x000004      
+ * 0x0256[23:0]  R4_REG               4              0x000004      
+ * 0x0259[23:0]  R5_REG               4              0x000004      
+ * 0x025C[23:0]  R6_REG               3              0x000003      
+ * 0x025F[23:0]  R7_REG               3              0x000003      
+ * 0x0262[23:0]  R8_REG               0              0x000000      
+ * 0x0268[23:0]  R9_REG               0              0x000000      
+ * 0x026B[7:0]   DESIGN_ID0           0              0x00          
+ * 0x026C[7:0]   DESIGN_ID1           0              0x00          
+ * 0x026D[7:0]   DESIGN_ID2           0              0x00          
+ * 0x026E[7:0]   DESIGN_ID3           0              0x00          
+ * 0x026F[7:0]   DESIGN_ID4           0              0x00          
+ * 0x0270[7:0]   DESIGN_ID5           0              0x00          
+ * 0x0271[7:0]   DESIGN_ID6           0              0x00          
+ * 0x0272[7:0]   DESIGN_ID7           0              0x00          
+ * 0x0302[43:0]  N0_NUM               30064771072    0x00700000000 
+ * 0x0308[31:0]  N0_DEN               2147483648     0x80000000    
+ * 0x030C[0]     N0_UPDATE            0              0x0           
+ * 0x030D[43:0]  N1_NUM               120259084288   0x01C00000000 
+ * 0x0313[31:0]  N1_DEN               2684354560     0xA0000000    
+ * 0x0317[0]     N1_UPDATE            0              0x0           
+ * 0x0318[43:0]  N2_NUM               0              0x00000000000 
+ * 0x031E[31:0]  N2_DEN               0              0x00000000    
+ * 0x0322[0]     N2_UPDATE            0              0x0           
+ * 0x0323[43:0]  N3_NUM               0              0x00000000000 
+ * 0x0329[31:0]  N3_DEN               0              0x00000000    
+ * 0x032D[0]     N3_UPDATE            0              0x0           
+ * 0x032E[43:0]  N4_NUM               0              0x00000000000 
+ * 0x0334[31:0]  N4_DEN               0              0x00000000    
+ * 0x0338[0]     N4_UPDATE            0              0x0           
+ * 0x0338[1]     N_UPDATE             0              0x0           
+ * 0x0339[4:0]   N_FSTEP_MSK          31             0x1F          
+ * 0x033B[43:0]  N0_FSTEPW            0              0x00000000000 
+ * 0x0341[43:0]  N1_FSTEPW            0              0x00000000000 
+ * 0x0347[43:0]  N2_FSTEPW            0              0x00000000000 
+ * 0x034D[43:0]  N3_FSTEPW            0              0x00000000000 
+ * 0x0353[43:0]  N4_FSTEPW            0              0x00000000000 
+ * 0x0359[15:0]  N0_DELAY             0              0x0000        
+ * 0x035B[15:0]  N1_DELAY             0              0x0000        
+ * 0x035D[15:0]  N2_DELAY             0              0x0000        
+ * 0x035F[15:0]  N3_DELAY             0              0x0000        
+ * 0x0361[15:0]  N4_DELAY             0              0x0000        
+ * 0x0802[15:0]  FIXREGSA0            0              0x0000        
+ * 0x0804[7:0]   FIXREGSD0            0              0x00          
+ * 0x090E[0]     XAXB_EXTCLK_EN       0              0x0           
+ * 0x090E[1]     XAXB_PDNB            1              0x1           
+ * 0x091C[2:0]   ZDM_EN               4              0x4           
+ * 0x0943[0]     IO_VDD_SEL           1              0x1           
+ * 0x0949[3:0]   IN_EN                0              0x0           
+ * 0x0949[7:4]   IN_PULSED_CMOS_EN    0              0x0           
+ * 0x094A[7:4]   INX_TO_PFD_EN        0              0x0           
+ * 0x0A02[4:0]   N_ADD_0P5            0              0x00          
+ * 0x0A03[4:0]   N_CLK_TO_OUTX_EN     3              0x03          
+ * 0x0A04[4:0]   N_PIBYP              1              0x01          
+ * 0x0A05[4:0]   N_PDNB               3              0x03          
+ * 0x0B44[3:0]   PDIV_ENB             15             0xF           
+ * 0x0B4A[4:0]   N_CLK_DIS            28             0x1C
+ * 
+ *
+ */
+
+#endif
\ No newline at end of file
--- a/board/tqc/common/tqc_moment_connect.c
+++ b/board/tqc/common/tqc_moment_connect.c
@@ -17,14 +17,26 @@
 #include <asm/arch/immap_lsch2.h>
 #endif
 #include <asm/mach-imx/mxc_i2c.h>
+#include <asm/arch/fsl_serdes.h>
 #include "tqc_moment_connect.h"
 #include "Si5341-RevA-Registers.h"
+#include "Si5341-RevB-Registers.h"
 #include "Si5341-RevD-Registers.h"
 
 
 static uint8_t clkgen_addr = 0;
 static uint8_t clkgen_current_page = 255;
 
+// Clock gen registers
+#define PAGE_REG		0x0001
+#define PART_NUMBER_REG		0x0002
+#define GRADE_REG		0x0004
+#define DEVICE_REV_REG		0x0005
+#define STATUS_REG		0x000C
+#define STICKY_STATUS_REG	0x0011
+#define INPUT_CLOCK_REG		0x0021
+
+
 struct tqc_moment_connect_i2c_gpio {
 	char name[20];
 	uint8_t addr;
@@ -58,7 +70,7 @@ struct tqc_moment_connect_i2c_gpio tqc_m
 #if 0
 int debug_return_request(void) {
 	printf("Press s to quit current function\n");
-	if (getc() == 's')
+	if (getchar() == 's')
 		return 1;
 
 	return 0;
@@ -198,7 +210,7 @@ static int clkgen_reg_write(uint16_t reg
 
 	if (page != clkgen_current_page) {
 		clkgen_current_page = page;
-		ret = i2c_write(clkgen_addr, 0x0001, 1, &page, 1);
+		ret = i2c_write(clkgen_addr, PAGE_REG, 1, &page, 1);
 		if (ret) return ret;
 	}
 
@@ -212,7 +224,7 @@ static int clkgen_reg_read(uint16_t reg,
 
 	if (page != clkgen_current_page) {
 		clkgen_current_page = page;
-		ret = i2c_write(clkgen_addr, 0x0001, 1, &page, 1);
+		ret = i2c_write(clkgen_addr, PAGE_REG, 1, &page, 1);
 		if (ret) return ret;
 	}
 
@@ -229,8 +241,13 @@ int tqc_moment_connect_clk_cfg_init(void
 	uint8_t found = 0;
 	si5341_revd_register_t * si5341_registers;
 	int si5341_registers_nb;
+	char si5341_rev = 0;
 	uint8_t preamble_size;
-	uint8_t val;
+
+	// Reconfigure clock only if PLL are not locked
+	printf("get_pll_status=%.02x\n", get_pll_status());
+	if (get_pll_status() == 0xf)
+		return 0;
 
 	oldbus = i2c_get_bus_num();
 	
@@ -244,7 +261,7 @@ int tqc_moment_connect_clk_cfg_init(void
 		if (i2c_probe(chip) == 0) {
 			printf("\tfound %d:%.02x\n", bus, chip);
 			found = 1;
-			if ((chip == SI5341B_REVA_CLKBUF_ADDR) || (chip == SI5341B_REVD_CLKBUF_ADDR))
+			if ((chip == SI5341B_REVA_CLKBUF_ADDR) || (chip == SI5341B_REVB_CUSTOM_CLKBUF_ADDR) || (chip == SI5341B_REVD_CLKBUF_ADDR))
 				clkgen_addr = chip;
 		}
 	}
@@ -255,54 +272,85 @@ int tqc_moment_connect_clk_cfg_init(void
 	printf("clkgen_addr=%.02x\n", clkgen_addr);
 	//i2c_enable_log();
 
+#if 0
+	i2c_set_bus_num(SI5341B_CLKBUF_BUS_NUM);
+
+	u8 part_number[2];
+	ret = clkgen_reg_read(PART_NUMBER_REG, (u8*)part_number);
+	if (ret) {
+		printf("Part number unreachable\n");
+		goto ERROR;
+	}
+	ret = clkgen_reg_read(PART_NUMBER_REG+1, (u8*)part_number+1);
+	if (ret) {
+		printf("Part number unreachable\n");
+		goto ERROR;
+	}
+	printf("part_number=%.02x%.02x\n", part_number[1], part_number[0]);
+
+	u8 grade;
+	ret = clkgen_reg_read(GRADE_REG, &grade);
+	if (ret) {
+		printf("Grade unreachable\n");
+		goto ERROR;
+	}
+	printf("grade=%.02x\n", grade);
+
+	u8 status;
+	ret = clkgen_reg_read(STATUS_REG, &status);
+	if (ret) {
+		printf("Status unreachable\n");
+		goto ERROR;
+	}
+	printf("status=%.02x\n", status);
+
+	u8 sticky_status;
+	ret = clkgen_reg_read(STICKY_STATUS_REG, &sticky_status);
+	if (ret) {
+		printf("Sticky status unreachable\n");
+		goto ERROR;
+	}
+	printf("sticky status=%.02x\n", sticky_status);
+#endif
+
+	i2c_set_bus_num(SI5341B_CLKBUF_BUS_NUM);
+
 	if (clkgen_addr == SI5341B_REVA_CLKBUF_ADDR) {
+		si5341_rev = 'A';
+
 		si5341_registers = (si5341_revd_register_t*)&si5341_reva_registers;
 		si5341_registers_nb = SI5341_REVA_REG_CONFIG_NUM_REGS;
 		preamble_size = 2;
+	} else if ((clkgen_addr == SI5341B_REVB_CUSTOM_CLKBUF_ADDR) || (clkgen_addr == SI5341B_REVD_CLKBUF_ADDR)) {
+		// Unless it has been customized, I2C address of REVB and REVD are the same. Distinction is done with DEVICE_REV internal register
+		u8 device_rev;
+
+		ret = clkgen_reg_read(DEVICE_REV_REG, &device_rev);
+		if (ret) {
+			printf("Device revision unreachable\n");
+			goto ERROR;
+		}
+		printf("Device revision : 0x%.02x\n", device_rev);
 		
-	} else if (clkgen_addr == SI5341B_REVD_CLKBUF_ADDR) {
-		si5341_registers = (si5341_revd_register_t*)&si5341_revd_registers;
-		si5341_registers_nb = SI5341_REVD_REG_CONFIG_NUM_REGS;
-		preamble_size = 6;
+		if (device_rev == 0x01) {
+			si5341_rev = 'B';
+			si5341_registers = (si5341_revd_register_t*)&si5341_revb_registers;
+			si5341_registers_nb = SI5341_REVB_REG_CONFIG_NUM_REGS;
+			preamble_size = 2;
+		} else if (device_rev >= 0x02) {
+			si5341_rev = 'D';
+			si5341_registers = (si5341_revd_register_t*)&si5341_revd_registers;
+			si5341_registers_nb = SI5341_REVD_REG_CONFIG_NUM_REGS;
+			preamble_size = 6;
+		} else {
+			goto ERROR;
+		}
 	} else {
 		goto ERROR;
 	}
 
-	i2c_set_bus_num(SI5341B_CLKBUF_BUS_NUM);
-
-#if 0
-	ret = i2c_read(TQC_MOMENT_CONNECT_I2C_CLKBUF_ADDR, 0x02, 1, buf, 2);
-	if(ret != 0)
-		printf("Clock buffer unreachable\n");
-	else
-		printf("Clock whoami : %.02x%.02x\n", buf[1], buf[0]); // 7f92 !
-	
-	ret = i2c_read(TQC_MOMENT_CONNECT_I2C_CLKBUF_ADDR, 0x0b, 1, buf, 1);
-	if(ret != 0)
-		printf("Clock buffer unreachable\n");
-	else
-		printf("I2C address : %.02x\n", buf[0]);
-	
-	ret = i2c_read(TQC_MOMENT_CONNECT_I2C_CLKBUF_ADDR, 0xe2, 1, buf, 1);
-	if(ret != 0)
-		printf("Clock buffer unreachable\n");
-	else
-		printf("Active NVM bank : %.02x\n", buf[0]);
-#endif
-
-	// Check if clocks already reconfigured
-	// read IN_SEL (IN_SEL_REGCTRL)
-	// Must be 0x07
-	ret = clkgen_reg_read(0x0021, &val);
-	if (ret) goto ERROR;
-	
-	if ((val & 0x7) == 0x7) {
-		// Already configured
-		goto SUCCESS;
-	}
-
 	// Dynamic PLL change (needed preamble)
-	if (clkgen_addr == SI5341B_REVA_CLKBUF_ADDR) {
+	if ((si5341_rev == 'A') || (si5341_rev == 'B')) {
 		ret = clkgen_reg_write(0x0B24, 0xD8);
 		if (ret) goto ERROR;
 		ret = clkgen_reg_write(0x0B25, 0x00);
@@ -349,7 +397,7 @@ SUCCESS:
 
 void tqc_moment_connect_xfi_init(struct ccsr_serdes *serdes, int lane)
 {
-	u32 gcr0, rec0;;
+	u32 gcr0;
 
 	//printf("## Press 's' to skip XFI reconfiguration\n");
 	//if (getc() == 's')
@@ -368,6 +416,7 @@ void tqc_moment_connect_xfi_init(struct
 
 	/* Disable Rx Equalization Boost */
 	/*
+	u32 rec0;
 	rec0 = serdes_in32(&serdes->lane[lane].rec0);
 	rec0 &= ~(1<<28);
 	serdes_out32(&serdes->lane[lane].rec0, rec0);
@@ -395,7 +444,7 @@ void tqc_moment_connect_lane_disable(str
 
 
 #ifndef CONFIG_SPL_BUILD
-static char *_srds_proto_name(int proto)
+static char * _srds_proto_name(int proto)
 {
 	switch(proto) {
 	case TQC_MOMENT_CONNECT_SRDS_PROTO_UNUSED:
@@ -420,7 +469,6 @@ static char *_srds_proto_name(int proto)
 		return "Unknown   ";
 	}
 }
-#endif
 
 static int _srds_mux_check_and_print(int port, int lane, int mux, int proto)
 {
@@ -476,4 +524,5 @@ int tqc_moment_connect_serdes_init(struc
 
 	return 0;
 }
+#endif
 
--- a/include/configs/tqc_bb_moment_connect.h
+++ b/include/configs/tqc_bb_moment_connect.h
@@ -25,8 +25,9 @@
 
 /* I2C clock buffer on baseboard (Si5341) */
 #define SI5341B_CLKBUF_BUS_NUM  0
-#define SI5341B_REVA_CLKBUF_ADDR     0x7c
-#define SI5341B_REVD_CLKBUF_ADDR     0x74
+#define SI5341B_REVA_CLKBUF_ADDR     	0x7c
+#define SI5341B_REVB_CUSTOM_CLKBUF_ADDR 0x34
+#define SI5341B_REVD_CLKBUF_ADDR     	0x74
 
 #endif /* __TQC_BB_MOMENT_CONNECT_H__ */
 
