--- a/board/tqc/common/tqc_moment_connect.c
+++ b/board/tqc/common/tqc_moment_connect.c
@@ -106,6 +106,7 @@ int tqc_moment_connect_i2c_gpios_init(vo
 	int i;
 	char *env;
 	uint8_t val;
+	int mb_type;
 
 	oldbus = i2c_get_bus_num();
 	i2c_set_bus_num(PCA953X_I2C_GPIO_BUS_NUM);
@@ -121,6 +122,19 @@ int tqc_moment_connect_i2c_gpios_init(vo
 			tqc_moment_connect_i2c_gpios[i].dir << tqc_moment_connect_i2c_gpios[i].nr);
 	}
 
+	// Get motherboard type
+	mb_type = tqc_moment_connect_i2c_gpio_get("led_bp");
+	if (mb_type == 0) {
+		printf("MB is a SERVER\n");
+		env_set("mb_type", "SERVER");
+	} else if (mb_type  == 1) {
+		printf("MB is a WAP\n");
+		env_set("mb_type", "WAP");
+	} else {
+		printf("Can't identify MB. Error %d\n", mb_type);
+		env_set("mb_type", "UNKNOWN");
+	}
+
 	// Reset PHY after power-up
 	printf("### Reset PHYs\n");
 	tqc_moment_connect_i2c_gpio_set("mrst1", 0);
