--- a/hal/dp_ops/edma_dp/edma_v2/edma.c
+++ b/hal/dp_ops/edma_dp/edma_v2/edma.c
@@ -1863,7 +1863,7 @@ static int edma_hang_recovery(void)
  * edma_hang_recovery_handler()
  *	to trigger recovery API for EDMA hang
  */
-int edma_hang_recovery_handler(struct ctl_table *table, int write,
+int edma_hang_recovery_handler(const struct ctl_table *table, int write,
                 void __user *buffer, size_t *lenp, loff_t *ppos)
 {
 	int ret;
--- a/hal/dp_ops/edma_dp/edma_v2/edma.h
+++ b/hal/dp_ops/edma_dp/edma_v2/edma.h
@@ -376,7 +376,7 @@ void edma_misc_stats_free(void);
 void edma_enable_interrupts(struct edma_gbl_ctx *egc);
 void edma_disable_interrupts(struct edma_gbl_ctx *egc);
 void edma_configure_rps_hash_map(struct edma_gbl_ctx *egc);
-int edma_hang_recovery_handler(struct ctl_table *table, int write, void __user *buffer, size_t *lenp, loff_t *ppos);
+int edma_hang_recovery_handler(const struct ctl_table *table, int write, void __user *buffer, size_t *lenp, loff_t *ppos);
 
 /*
  * edma_reg_read()
--- a/hal/dp_ops/edma_dp/edma_v2/edma_cfg_rx.c
+++ b/hal/dp_ops/edma_dp/edma_v2/edma_cfg_rx.c
@@ -1517,7 +1517,7 @@ void edma_cfg_rx_napi_add(struct edma_gb
  * Similarly, whenever the free Rx ring descriptor count crosses the X-ON value,
  * the ring level flow control will be disabled.
  */
-int edma_cfg_rx_fc_enable_handler(struct ctl_table *table, int write,
+int edma_cfg_rx_fc_enable_handler(const struct ctl_table *table, int write,
 		void __user *buffer, size_t *lenp, loff_t *ppos)
 {
 	int ret;
@@ -1575,7 +1575,7 @@ int edma_cfg_rx_fc_enable_handler(struct
  * edma_cfg_rx_queue_tail_drop_handler()
  *	API to enable/disable queue tail drop configuration
  */
-int edma_cfg_rx_queue_tail_drop_handler(struct ctl_table *table, int write,
+int edma_cfg_rx_queue_tail_drop_handler(const struct ctl_table *table, int write,
 		void __user *buffer, size_t *lenp, loff_t *ppos)
 {
 	int ret;
@@ -1622,7 +1622,7 @@ int edma_cfg_rx_queue_tail_drop_handler(
  * edma_cfg_rx_rps()
  *	API to configure RPS hash mapping for the given number of cores
  */
-int edma_cfg_rx_rps(struct ctl_table *table, int write,
+int edma_cfg_rx_rps(const struct ctl_table *table, int write,
 		void __user *buffer, size_t *lenp, loff_t *ppos)
 {
 	int ret;
@@ -1659,7 +1659,7 @@ int edma_cfg_rx_rps(struct ctl_table *ta
  * us to distinguish between rps_num_core and rps_bitmap_core as both
  * are calling the same configure hash function.
  */
-int edma_cfg_rx_rps_bitmap(struct ctl_table *table, int write,
+int edma_cfg_rx_rps_bitmap(const struct ctl_table *table, int write,
 		void __user *buffer, size_t *lenp, loff_t *ppos)
 {
 	int ret;
--- a/hal/dp_ops/edma_dp/edma_v2/edma_cfg_rx.h
+++ b/hal/dp_ops/edma_dp/edma_v2/edma_cfg_rx.h
@@ -67,13 +67,13 @@ void edma_cfg_rx_rings_enable(struct edm
 void edma_cfg_rx_rings_disable(struct edma_gbl_ctx *egc);
 void edma_cfg_rx_ring_reset(struct edma_rxdesc_ring *ring);
 bool edma_cfg_rx_ring_en_mapped_queues(struct edma_gbl_ctx *egc, uint16_t ring_id, bool enable);
-int edma_cfg_rx_fc_enable_handler(struct ctl_table *table, int write,
+int edma_cfg_rx_fc_enable_handler(const struct ctl_table *table, int write,
 		void __user *buffer, size_t *lenp, loff_t *ppos);
 void edma_cfg_rx_page_mode_and_jumbo(struct edma_gbl_ctx *egc);
-int edma_cfg_rx_queue_tail_drop_handler(struct ctl_table *table, int write,
+int edma_cfg_rx_queue_tail_drop_handler(const struct ctl_table *table, int write,
 		void __user *buffer, size_t *lenp, loff_t *ppos);
-int edma_cfg_rx_rps(struct ctl_table *table, int write,
+int edma_cfg_rx_rps(const struct ctl_table *table, int write,
 		void __user *buffer, size_t *lenp, loff_t *ppos);
-int edma_cfg_rx_rps_bitmap(struct ctl_table *table, int write,
+int edma_cfg_rx_rps_bitmap(const struct ctl_table *table, int write,
 		void __user *buffer, size_t *lenp, loff_t *ppos);
 #endif	/* __EDMA_CFG_RX_H__ */
