--- a/include/iwinfo.h
+++ b/include/iwinfo.h
@@ -30,6 +30,7 @@
 #define IWINFO_80211_AC      (1 << 4)
 #define IWINFO_80211_AD      (1 << 5)
 #define IWINFO_80211_AX      (1 << 6)
+#define IWINFO_80211_COUNT   7
 
 #define IWINFO_BAND_24       (1 << 0)
 #define IWINFO_BAND_5        (1 << 1)
@@ -71,7 +72,7 @@
 extern const char *IWINFO_CIPHER_NAMES[IWINFO_CIPHER_COUNT];
 extern const char *IWINFO_KMGMT_NAMES[IWINFO_KMGMT_COUNT];
 extern const char *IWINFO_AUTH_NAMES[IWINFO_AUTH_COUNT];
-
+extern const char *IWINFO_80211_NAMES[IWINFO_80211_COUNT];
 
 enum iwinfo_opmode {
 	IWINFO_OPMODE_UNKNOWN    = 0,
--- a/iwinfo_lib.c
+++ b/iwinfo_lib.c
@@ -22,6 +22,16 @@
 /*
  * name constants
  */
+const char *IWINFO_80211_NAMES[IWINFO_80211_COUNT] = {
+       "a",
+       "b",
+       "g",
+       "n",
+       "ac",
+       "ad",
+       "ax",
+};
+
 const char *IWINFO_CIPHER_NAMES[] = {
 	"NONE",
 	"WEP40",
